easypy/concurrency.py

```diff
@@ -933,12 +933,19 @@ def __init__(self, func, *args, **kwargs):
         self.kwargs = kwargs
         self.throw = kwargs.pop('throw', True)
         self.daemon = kwargs.pop('daemon', True)
-        self.threadname = kwargs.pop('threadname', 'anon-%X' % id(self))
         self.stopper = kwargs.pop('stopper', threading.Event())
         self.sleep = kwargs.pop('sleep', 1)
         self.loop = kwargs.pop('loop', False)
         self.timer = None
         self.console_logging = kwargs.pop('console_logging', True)
+        self.threadname = kwargs.pop('threadname', None)
+        if not self.threadname:
+            current_thread_name = threading.current_thread().name
+            if current_thread_name:
```

Comment:
is there a thread that _doesn't_ have a name?
I think every `concurrent` call is going to have a `MainThread::` prefix...

Reply:
Don't remember seeing something like this, but wanted to be on the safe side.

Reply:
Well I think the `MainThread::` prefix is redundant and shouldn't be there. A `concurrent` thread spawned by the main thread should just have its `anon-xxx` name with no prefix.
would you fix that for us?